### ISSUE TITLE
 Sizes of tensors must match except in dimension 1

### ISSUE THREAD

**Issue by Laidawang** [UCSB-NLP-Chang/DiffSTE#3]
when i use:
python generate.py --ckpt_path weights/diffste.ckpt --in_image examples/sample1.png --in_mask examples/mask1.png --text five --font Courgette --color red --output_dir output
i got Sizes of tensors must match except in dimension 1. Expected size 64 but got size 96 for tensor number 1 in the list.

│ /data2/home/srchen/project/github/DiffSTE/src/trainers/inpaint_trainer.py:548 in sample_loop     │
│                                                                                                  │
│   545 │   │   │   │   │   [latents] * 2) if do_classifier_free_guidance else latents             │
│   546 │   │   │   )                                                                              │
│   547 │   │   │                                                                                  │
│ ❱ 548 │   │   │   latent_model_input = torch.cat(                                                │
│   549 │   │   │   │   [latent_model_input, mask, masked_image_latents], dim=1                    │
│   550 │   │   │   )                                                                              │
│   551 │   │   │   latent_model_input = self.noise_scheduler.scale_model_input(

**Comment by Laidawang** [comment]
I checked the code and it seems to be a non-square problem,
![image](https://user-images.githubusercontent.com/85244566/233826436-3b107528-dd39-4439-8887-fe241ee4bebb.png)

But I don't know if this will have any other effect

**Comment by Question406** [comment]
Hi, the image should be square, as the diffusion model only takes a square image as input.
